app/models/grda_warehouse/sex_selector.rb

```diff
@@ -0,0 +1,158 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module GrdaWarehouse
+  # Determines the preferred Sex value from a set of source client records.
+  # This service object encapsulates the business rules for selecting the best
+  # available Sex value based on a multi-level sorting algorithm.
+  #
+  # The selection process prioritizes candidates based on the following criteria
+  # in order:
+  #   1. Sex value preference (0 or 1 preferred over 8, 9, 99)
+  #   2. Record timestamp (DateUpdated), preferring most recent
+  #   3. Record ID (lower is better)
+  #
+  # Valid Sex values:
+  #   0 => Female
+  #   1 => Male
+  #   8 => Client doesn't know
+  #   9 => Client prefers not to answer
+  #   99 => Data not collected
+  #
+  # This class handles data normalization for various source data formats
+  # (e.g., Hashes, ActiveRecord objects). The goal is to provide a consistent
+  # and reliable way to determine the most trustworthy Sex value from potentially
+  # conflicting source data.
+  class SexSelector
+    Candidate = Struct.new(:value, :tie_breakers, keyword_init: true)
+
+    VALID_PRIORITIZATION_METHODS = [:newest_first].freeze
+
+    def self.call(dest_attr:, source_clients:, prioritization_method: :newest_first)
+      new(dest_attr: dest_attr, source_clients: source_clients, prioritization_method: prioritization_method).call
+    end
+
+    def initialize(dest_attr:, source_clients:, prioritization_method: :newest_first)
+      @dest_attr = dest_attr.with_indifferent_access
+      @source_clients = Array.wrap(source_clients).map { |client| normalize_source(client) }
+      @prioritization_method = prioritization_method
+
+      return if VALID_PRIORITIZATION_METHODS.include?(@prioritization_method)
+
+      raise ArgumentError, "Invalid prioritization_method: #{@prioritization_method}. Valid methods: #{VALID_PRIORITIZATION_METHODS.join(', ')}"
```

Comment:
Might be easier to read:
```suggestion
      raise ArgumentError, "Invalid prioritization_method: #{@prioritization_method}. Valid methods: #{VALID_PRIORITIZATION_METHODS.join(', ')}" unless VALID_PRIORITIZATION_METHODS.include?(@prioritization_method)
```